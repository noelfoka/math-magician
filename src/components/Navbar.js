import { Outlet, Link } from 'react-router-dom';

const Navbar = () => (
  <>

    <nav className="navbar">
      <h1>Math Magicians</h1>
      <ul>
        <li>
          <Link to="/math-magicians">Home</Link>
        </li>
        <li>
          <Link to="/math-magicians/Calculator">Calculator</Link>
        </li>
        <li>
          <Link to="/math-magicians/Quote">Quote</Link>
        </li>
      </ul>
    </nav>

    <Outlet />
  </>
);

export default Navbar;
