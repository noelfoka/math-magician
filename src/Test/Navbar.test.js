import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

it('renders correctly', () => {
  const nav = renderer
    .create(
      <Router>
        <Navbar />
      </Router>,
    )
    .toJSON();
  expect(nav).toMatchSnapshot();
});
