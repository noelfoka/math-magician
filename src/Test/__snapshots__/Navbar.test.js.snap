// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navbar"
>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        href="/math-magicians"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/math-magicians/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/math-magicians/Quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
